### PR TITLE
[Compute] vm create: Automatic in-guest patching

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

--enable-auto-update
Resolve https://github.com/Azure/azure-cli/issues/14458
Indicate whether Automatic Updates is enabled for the Windows virtual machine
--patch-mode
Mode of in-guest patching to IaaS virtual machine. Possible values are: Manual - You  control the application of patches to a virtual machine. You do this by applying patches manually inside the VM. In this mode, automatic updates are disabled; the property WindowsConfiguration.enableAutomaticUpdates must be false. AutomaticByOS - The virtual machine will automatically be updated by the OS. The property WindowsConfiguration.enableAutomaticUpdates must be true. AutomaticByPlatform - the virtual machine will automatically updated by the OS. The properties provisionVMAgent and WindowsConfiguration.enableAutomaticUpdates must be true

**Testing Guide**  

```
az vm create -g rg -n vm --image Win2019Datacenter --enable-agent --enable-auto-update --patch-mode AutomaticByOS --admin-username azureuser --admin-password testPassword0 --nsg-rule NONE')
```

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
vm create

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -330,6 +330,10 @@ def load_arguments(self, _):
                        help="The eviction policy for the Spot priority virtual machine. Default eviction policy is Deallocate for a Spot priority virtual machine")
         c.argument('enable_agent', arg_type=get_three_state_flag(), min_api='2018-06-01',
                    help='Indicates whether virtual machine agent should be provisioned on the virtual machine. When this property is not specified, default behavior is to set it to true. This will ensure that VM Agent is installed on the VM so that extensions can be added to the VM later')
+        c.argument('enable_auto_update', arg_type=get_three_state_flag(), min_api='2020-06-01',
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:333]: is this a windows only property in Swagger?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -330,6 +330,10 @@ def load_arguments(self, _):
...
+                   help='Indicate whether Automatic Updates is enabled for the Windows virtual machine')
+        c.argument('patch_mode', arg_type=get_enum_type(self.get_models('InGuestPatchMode')), min_api='2020-06-01',
+                   help='Mode of in-guest patching to IaaS virtual machine. Possible values are: Manual - You  control the application of patches to a virtual machine. You do this by applying patches manually inside the VM. In this mode, automatic updates are disabled; the property WindowsConfiguration.enableAutomaticUpdates must be false. AutomaticByOS - The virtual machine will automatically be updated by the OS. The property WindowsConfiguration.enableAutomaticUpdates must be true. AutomaticByPlatform - the virtual machine will automatically updated by the OS. The properties provisionVMAgent and WindowsConfiguration.enableAutomaticUpdates must be true')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:336]: how about to translater `WindowsConfiguration.enableAutomaticUpdates` to CLI command argument?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:333]: checked Swagger, answer is yes.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:336]: OK

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 3]
```diff
@@ -300,6 +300,14 @@ def _build_os_profile():
         if secrets:
             os_profile['secrets'] = secrets
 
+        if enable_auto_update is not None and custom_image_os_type.lower() == 'windows':
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:303]: How about combine these 2 `custom_image_os_type.lower() == 'windows'` ?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:303]: I group code by parameter. It is not a serious problem.

**Pull request merged**